src/BlocHost-Morphic/BlMorphicHostSpace.class.st

```diff
@@ -123,7 +123,12 @@ BlMorphicHostSpace >> physicalSize [
 	This size may differ from the logical size on high dpi (retina) screens.
 	In most cases physical size is x2 larger than logical size on retina screens."
 
-	^ (spaceHostMorph width @ spaceHostMorph height) asPhysicalSize
+	| canvasScaleFactor |
+
+	canvasScaleFactor := SystemVersion current major >= 12
+		ifTrue: [ (Smalltalk at: #OSWorldRenderer) canvasScaleFactor ]
```

Comment:
I think `OSWorldRenderer` existed in Pharo for several years, in which case the "Smalltalk at:" would not be needed... or is it a new class name?

Reply:
Getting the class through Smalltalk like this avoids an ‘unknown variable’ error in versions of Pharo that don’t have the class, which is probably not really relevant here as OSWorldRenderer exists since Pharo 8, but it also avoids a match of ReSendsUnknownMessageToGlobalRule in Pharo 11.